overlord/snapstate/backend/setup.go

```diff
@@ -236,8 +238,40 @@ func (b Backend) RemoveSnapFiles(s snap.PlaceInfo, typ snap.Type, installRecord
 	return nil
 }
 
+// RemoveComponentOpts are options considered when removing a component.
+type RemoveComponentOpts struct {
+	// MaybeInitramfsMounted is set if the component was mounted also from
+	// the initramfs, which can be the case for kernel-modules components.
+	MaybeInitramfsMounted bool
+}
+
 // RemoveComponentFiles unmounts and removes component files from the disk.
-func (b Backend) RemoveComponentFiles(cpi snap.ContainerPlaceInfo, installRecord *InstallRecord, dev snap.Device, meter progress.Meter) error {
+func (b Backend) RemoveComponentFiles(cpi snap.ContainerPlaceInfo, installRecord *InstallRecord, dev snap.Device, opts RemoveComponentOpts, meter progress.Meter) error {
+	if opts.MaybeInitramfsMounted {
+		// Stop unit created from initramfs for kernel-modules components, if
+		// existing (if we have not rebooted after installation it will not
+		// be there either). Note that this mount can exist only on UC, on
+		// hybrid the initramfs will create the mount exactly in the same place
+		// as snapd, so there is no duplication.
+		mntPoint := filepath.Join(dirs.GlobalRootDir, "writable", "system-data",
+			dirs.StripRootDir(cpi.MountDir()))
+		isMounted, err := osutil.IsMounted(mntPoint)
+		if err != nil {
+			return err
+		}
+		// We do not use systemd as there is no associated unit file - the
+		// unit file created by the initramfs has a "sysroot-" prefix to the
+		// real mount path, so systemd does not consider it associated with
+		// the mount. This unit file is inactive therefore. We leave it as it
+		// is, it will disappear in next reboot and it would be a waste to
+		// remove it and do a daemon-reload.
+		if isMounted {
+			if output, err := exec.Command("umount", "--lazy", mntPoint).CombinedOutput(); err != nil {
+				return osutil.OutputErr(output, err)
+			}
```

Comment:
there is code to do umount --lazy also in preesed, wondering if we should make a osutil helper?